hazelcast-hibernate5/src/main/java/com/hazelcast/hibernate/telemetry/PhoneHomeService.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.hazelcast.hibernate.telemetry;
+
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import java.io.BufferedInputStream;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.time.Duration;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.hazelcast.internal.nio.IOUtil.closeResource;
+import static java.lang.Boolean.FALSE;
+import static java.lang.System.getenv;
+import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
+
+/**
+ * Pings phone home server with plugin information daily.
+ *
+ * @since 2.1.2
+ */
+public class PhoneHomeService {
+
+    private static final String SYS_PHONE_HOME_ENABLED = "hazelcast.phone.home.enabled";
+    private static final String ENV_PHONE_HOME_ENABLED = "HZ_PHONE_HOME_ENABLED";
+
+    private static final String PHONE_HOME_URL = "http://phonehome.hazelcast.com/pingIntegrations/hazelcast-hibernate5";
+    private static final Duration TIMEOUT = Duration.ofMillis(3000);
+    private static final int RETRY_COUNT = 5;
+
+    private final ILogger logger = Logger.getLogger(PhoneHomeService.class);
+    private final AtomicBoolean started = new AtomicBoolean();
+    private final ScheduledExecutorService executor;
+
+    private PhoneHomeInfo phoneHomeInfo;
+
+    public PhoneHomeService() {
+        executor = newSingleThreadScheduledExecutor(r -> {
+            Thread t = new Thread(r, "PhoneHomeService");
+            t.setDaemon(true);
+            return t;
+        });
+    }
+
+    public void start() {
+        if (started.compareAndSet(false, true) && isPhoneHomeEnabled()) {
+            executor.scheduleAtFixedRate(this::send, 0, 1, TimeUnit.DAYS);
+        }
+    }
+
+    private static boolean isPhoneHomeEnabled() {
+        if (FALSE.toString().equalsIgnoreCase(System.getProperty(SYS_PHONE_HOME_ENABLED))) {
```

Comment:
It would be probably safer to delegate the job to `Boolean` (although it probably does something similar underneath, but at least we wouldn't need to worry about it):

```suggestion
        if (FALSE == Boolean.parseBoolean(System.getProperty(SYS_PHONE_HOME_ENABLED))) {
```

Reply:
I afraid `FALSE == Boolean.parseBoolean(System.getProperty(SYS_PHONE_HOME_ENABLED))` (or its env version) holds when no system property (or env var) is set at all. Say someone hasn't set any sys property or env var, phone calls are going to be disabled with the updated setup until users explicitly enable this. I think this should be enabled by default and can be disabled explicitly. WDYT?

Reply:
Agreed, missed that.

Try this then: `FALSE == Boolean.parseBoolean(System.getProperty(SYS_PHONE_HOME_ENABLED, "true"))`

Reply:
Any values, typos, etc. other than `true` will prevent calls. I believe we should check if users explicitly set one of them to nothing but `false` and otherwise we send calls. I think we can keep string checks unless you strongly disagree with the idea.